### PR TITLE
Implement 'maxBy', 'minBy', 'reject', 'all' and 'any' functions

### PR DESCRIPTION
**Pull request by lucasferrari07**
- Implement maxBy and minBy functions, which return the largest and
smallest element of an array which is tested by applying a function.
- Implement reject function which filters an array by applying a
function to each element and testing if it is false.
- Implement 'all' function which returns true if all elements in the
input array meet a condition
- Implement 'any' function which returns true if at least one element in
the input array meets a condition

### CODE REVIEW

File: __tests__/main.spec.js [hunk 1]
```diff
@@ -41,17 +41,17 @@ describe('reduce', () => {
 });
 
 describe('filter', () => {
-  test('filters and array of numbers by only even', () => {
+  test('filters an array of numbers by only even', () => {
```
Comment (jfarmer) [inline __tests__/main.spec.js]: Make sure to test corner cases:

- What do you want to happen when your input array is empty `[]`?
- What do you want to happen when your input array is not an array?
